lib/plugins/bayesian-optimization/presentation/dialogs/bayesian_optimization_training_dialog_bloc.dart

```diff
@@ -228,34 +226,30 @@ class BOTrainingDialogBloc
     on<DesiredBooleanValueUpdated>(_onDesiredBooleanValueUpdated);
   }
 
-  void _onDatasetSelected(
-      DatasetSelected event, Emitter<BOTrainingDialogState> emit) {
+  void _onDatasetSelected(DatasetSelected event, Emitter<BOTrainingDialogState> emit) {
     var availableFeatures = [''];
     if (event.dataset.toString() == 'Protein') {
       final ProteinRepository? biocentralDatabase =
-          _biocentralDatabaseRepository.getFromType(Protein)
-              as ProteinRepository?;
-      availableFeatures = biocentralDatabase!.getTrainableColumnNames();
+          _biocentralDatabaseRepository.getFromType(Protein) as ProteinRepository?;
+      availableFeatures = biocentralDatabase!.getPartiallyUnlabeledColumnNames();
     }
+
     emit(state.copyWith(
       selectedDataset: event.dataset,
       currentStep: BOTrainingDialogStep.taskSelection,
       availableFeatures: availableFeatures,
     ));
   }
 
-  void _onTaskSelected(
-      TaskSelected event, Emitter<BOTrainingDialogState> emit) {
-    final ProteinRepository? biocentralDatabase = _biocentralDatabaseRepository
-        .getFromType(Protein) as ProteinRepository?;
+  void _onTaskSelected(TaskSelected event, Emitter<BOTrainingDialogState> emit) {
+    final ProteinRepository? biocentralDatabase =
+        _biocentralDatabaseRepository.getFromType(Protein) as ProteinRepository?;
 
     List<String> filteredFeatures = [];
     if (event.task.contains('highest probability')) {
```

Comment:
As above, try to avoid strings because they are super error prone. Trying using explicit checks for a `Type` or an `Enum`.